lib/bolt/util.rb

```diff
@@ -43,6 +43,21 @@ def deep_merge(hash1, hash2)
         hash1.merge(hash2, &recursive_merge)
       end
 
+      # Accepts a Data object and returns a copy with all hash keys
```

Comment:
+9000. We keep needing then not needing this. Generalized method makes so much sense.